sklearn/ensemble/tests/test_gradient_boosting.py

```diff
@@ -1382,7 +1456,7 @@ def test_early_stopping_n_classes():
     X = [[1]] * 10
     y = [0, 0] + [1] * 8  # only 2 negative class over 10 samples
     gb = GradientBoostingClassifier(
-        n_iter_no_change=5, random_state=0, validation_fraction=8
+        n_iter_no_change=5, random_state=0, validation_fraction=0.8
```

Comment:
This is problematic here. Let me summarize why.

We document that `validation_fraction` should be a float and it is expected to be in [0, 1]. However, this test shows that we can accept integers. I was checking the code and the reason is that we use `train_test_split` that accepts both float and int.

So we have 2 choices here:

- We consider it as a bug fix. We will need an entry in what's new to acknowledge this fix.
- We need to keep backward compatibilities in which case we need to update the documentation and add a proper test to chcek that passing integer is behaving as expected.

Reply:
ping @thomasjpfan @ogrisel on this issue.

Reply:
I would say it is a bug fix. There shouldn't be too much of an issue given the documentation clearly states `[0, 1]` and the name of the parameter contains "fraction".

Reply:
OK, it works for me. We only need a changelog entry then.